src/imageops/sample.rs

```diff
@@ -261,10 +261,10 @@ where
 
             let (t1, t2, t3, t4) = (t.0 / sum, t.1 / sum, t.2 / sum, t.3 / sum);
             let t = Pixel::from_channels(
-                NumCast::from(clamp(t1, 0.0, max)).unwrap(),
-                NumCast::from(clamp(t2, 0.0, max)).unwrap(),
-                NumCast::from(clamp(t3, 0.0, max)).unwrap(),
-                NumCast::from(clamp(t4, 0.0, max)).unwrap(),
+                NumCast::from(clamp(t1, 0.0, max).round()).unwrap(),
+                NumCast::from(clamp(t2, 0.0, max).round()).unwrap(),
+                NumCast::from(clamp(t3, 0.0, max).round()).unwrap(),
+                NumCast::from(clamp(t4, 0.0, max).round()).unwrap(),
```

Comment:
This is going to be wrong if S is a floating point type. I'm not entirely sure how to resolve that